### ISSUE TITLE
How to achieve good performance on Object Detection model with ~40,000 training images

### ISSUE THREAD

**Issue by EthanV431** [ultralytics/ultralytics#20105]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I'm looking for ways to improve my object detection model. I'm currently training yolov11l for 100 epochs on the full dataset after finding some better hyperparameters via tuning. Are there any specific techniques I should keep in mind while training the model?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @EthanV431, thank you for reaching out and for your interest in Ultralytics 🚀! We recommend exploring the [Docs](https://docs.ultralytics.com/) where you can find detailed guidance on [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage as well as tips for optimizing your model training.

If this is a custom training ❓ question, we encourage you to provide as much information as possible to help us assist you better. This may include:
- Sample dataset images or a description of your dataset
- Training logs
- Details about your training parameters and configuration

Also, please ensure you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to maximize your model's performance.

Join the Ultralytics community to share knowledge or seek advice:
- For real-time chat, visit our [Discord](https://discord.com/invite/ultralytics) 🎧
- For in-depth discussions, check out [Discourse](https://community.ultralytics.com/)
- For community-driven insights, join our [Subreddit](https://reddit.com/r/Ultralytics)

## Upgrade

Please ensure you're on the latest version of the `ultralytics` package to benefit from the newest features and fixes. Upgrade using the following command:

```bash
pip install -U ultralytics
```

This ensures compatibility with the latest [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/).

## Environments

YOLO can be run in any of the following up-to-date verified environments with all dependencies preinstalled:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. These tests validate the correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across macOS, Windows, and Ubuntu.

This is an automated response to guide you in the right direction, but an Ultralytics engineer will also assist you soon. Let us know if you have additional details to share or questions to clarify! 😊

**Comment by glenn-jocher** [comment]
Great to hear you're working with YOLO11l on such a substantial dataset. Since you've already handled hyperparameter tuning, consider these additional techniques:

For small objects, try image tiling - dividing larger images into smaller segments preserves resolution and improves detection of tiny objects. 

When fine-tuning, set `warmup_epochs=0` to start with a higher learning rate since you're continuing from established weights.

Monitor key metrics during validation like mAP, IoU, precision, and recall to understand where your model might be struggling.

If you're dealing with class imbalance in your dataset, adjusting class weights during training might help improve performance for underrepresented classes.

The [YOLO Performance Metrics guide](https://docs.ultralytics.com/guides/yolo-performance-metrics/) offers additional insights on interpreting and improving results if you need more specific guidance.

**Comment by EthanV431** [comment]
How do I know if the hyperparameters that got chosen are good? I'm running the training script for 100 epochs and I'm not sure it's working. Here's the command i'm using:
!yolo detect train model="runs/detect/4-9-2025model3.0/weights/last.pt" data="data.yaml" multi_scale=True save=True save_period=1 name="4-9-2025model3.0" resume=True plots=true device=[0,1,2,3] batch=32 cache=True bgr=0.36825066993998545 box=0.12727664616028986 cls=0.6574567020755611 copy_paste=0.33243325588993433 degrees=12.180389338590011 fliplr=0.7729738137244045 flipud=0.3909211874502462 hsv_h=0.009537866794789775 hsv_s=0.27233772849420695 hsv_v=0.1368599352364378 lr0=0.09165172629321003 lrf=0.8926533417572727 mixup=0.7408115628335868 momentum=0.8447196896987875 mosaic=0.30457674812785407 perspective=0.0006683342142527883 scale=0.4099910321139165 shear=2.170543554469593 translate=0.43053892025105317 warmup_epochs=0.5828217461561414 warmup_momentum=0.49799507646685337 weight_decay=0.0003389265924935613

Here's the data from the last few epochs:

![Image](https://github.com/user-attachments/assets/9619042b-5f3c-40a8-b157-39814c35205d)

**Comment by glenn-jocher** [comment]
Based on the metrics you've shared, your model doesn't appear to be overfitting yet. The validation losses (val/box_loss, val/cls_loss, val/dfl_loss) are stable without increasing trends, which would be the primary sign of overfitting. Your mAP50-95 is holding steady around 0.61-0.62, which is quite good.

To evaluate if your hyperparameters are effective, look at these indicators:

1. Stable or improving validation metrics in later epochs
2. Good balance between precision (~0.9) and recall (~0.7-0.8)
3. Reasonable convergence rate (not oscillating wildly)

Your current setup looks promising, but if you're concerned about potential overfitting in later epochs, consider:

1. Implementing early stopping with patience (you can set this with the `patience` parameter)
2. Using `rect=true` for evaluation to handle variable image sizes effectively
3. Monitoring the gap between training and validation metrics - a widening gap indicates overfitting

If you want to extract specific metrics for deeper analysis, you can use `model.val()` results as shown in the model evaluation documentation to track specific performance metrics over time.
